### ISSUE TITLE
bors retry race condition

### ISSUE THREAD

**Issue by hackaugusto** [bors-ng/bors-ng#434]
`bors retry` will update the staging branch even if there is a build currently running. If a job has not started yet the expected commit will be deleted and the build will fail.

[job started before retry](https://travis-ci.org/raiden-network/raiden/jobs/394588783)
[`bors retry`](https://github.com/raiden-network/raiden/pull/1583#issuecomment-398778838)
[job started after retry](https://travis-ci.org/raiden-network/raiden/jobs/394588784)

**Comment by andschwa** [comment]
This appears to happen with lots of `bors try` in a row too.

It seems that as soon as Bors sees that a `bors try` build has started (but not finished), it thinks it's free to force-push the `trying` branch. However, our build system (Jenkins / Blue Ocean) will start the job as soon as it can, though sub-jobs may still be waiting for executors to become available. That is, the `trying` branch is used by Jenkins to checkout the repo throughout the job's life, and really isn't free to be force-pushed until the job is completed.

This seems related, but maybe it warrants a new issue.

**Comment by andschwa** [comment]
Our CI system is weighed down to the point that we had to stop using `bors try` since this happened every time, so we're stuck using the sandbox when testing CI changes. I do hope this gets fixed (though I'm not sure what the fix is as the issue truly lies in Jenkins).

**Comment by pietrocaselani** [comment]
We are facing the same problem using `bors try`. Any updates on this? For now, I think we will stop using `bors try`

**Comment by notriddle** [comment]
The commenters in here are talking about two issues.

One of them, the `bors try` race issue, is now fixed.

The other one, where bors r+ followed by a cancel followed by another job (either another r+ or just the next queued item running) caused Travis to start throwing up errors, is not a bug. It's caused by the fact that bors doesn't wait for canceled builds to be finished at the provider end before starting the next one. Since it's canceled at bors' end, I don't see why it matters that it's not able to finish. Won't fix.
